sdk/rust/src/processor/mod.rs

```diff
@@ -123,12 +129,51 @@ impl<'a> TransactionProcessor<'a> {
         true
     }
 
+    fn unregister(&mut self, mut sender: ZmqMessageSender) {
+        let request = TpUnregisterRequest::new();
+        info!("sending TpUnregisterRequest");
+        let serialized = match request.write_to_bytes() {
+            Ok(serialized) => serialized,
+            Err(err) => {
+                error!("Serialization failed: {}", err.description());
+                return
+            }
+        };
+        let x : &[u8] = &serialized;
+
+        let mut future = match sender.send(
+            Message_MessageType::TP_UNREGISTER_REQUEST,
+            &generate_correlation_id(),
+            x) {
+                Ok(fut) => fut,
+                Err(err) => {
+                    error!("Unregistration failed: {}", err.description());
+                    return
+                }
+            };
+        // Absorb the TpUnregisterResponse message
+        let _ = match future.get_timeout(Duration::from_millis(40000)){
```

Comment:
In the Python SDK, we only wait 1 second for this response.

Reply:
updated to 1 second